PR: https://github.com/odoo/odoo/pull/253079

From: 3259a7fae4ca4ee98f7e02da5109c43cd9e77f69
From: Tatiana Skvortsova (taskv)
Date: 2026-04-13 06:36:16

Breaking data model changes scores: del:4 + add:0, change matches:
-    departure_do_unassign_company_car = fields.Boolean(related='version_id.departure_do_unassign_company_car',
-    do_unassign_company_car = fields.Boolean(
-    _inherit = 'hr.version'
-    departure_do_unassign_company_car = fields.Boolean(related="departure_id.do_unassign_company_car")

Total Changes: 21

[IMP] l10n_be_hr_payroll: Improve End of Collaboration UX

Improve the UX of the "End of Collaboration" procedure, particularly for Belgium.

The option to select which actions are executed upon employee departure
(delete user, archive employee, unassign car, etc.) has been removed.

All actions are now executed automatically. However, it is still possible
to define the date on which they will be applied.

Related PRs:
Enterprise: odoo/enterprise#110147
Upgrade: odoo/upgrade#9653

task-5958676

test

closes odoo/odoo#253079

Signed-off-by: Yannick Tivisse (yti) <yti@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_fleet/models/__init__.py
+++ b/addons/hr_fleet/models/__init__.py
@@ -2,7 +2,6 @@
 
 from . import hr_employee
 from . import hr_employee_departure
-from . import hr_version
 from . import fleet_vehicle_assignation_log
 from . import fleet_vehicle
 from . import fleet_vehicle_log_contract

--- a/addons/hr_fleet/models/hr_employee.py
+++ b/addons/hr_fleet/models/hr_employee.py
@@ -14,8 +14,6 @@ class HrEmployee(models.Model):
         groups="fleet.fleet_group_manager,hr.group_hr_user",
     )
     license_plate = fields.Char(compute="_compute_license_plate", search="_search_license_plate", groups="hr.group_hr_user")
-    departure_do_unassign_company_car = fields.Boolean(related='version_id.departure_do_unassign_company_car',
-        inherited=True, readonly=False, groups="hr.group_hr_user")
 
     def action_open_employee_cars(self):
         self.ensure_one()

--- a/addons/hr_fleet/models/hr_employee_departure.py
+++ b/addons/hr_fleet/models/hr_employee_departure.py
@@ -1,19 +1,14 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import fields, models
+from odoo import models
 
 
 class HrEmployeeDeparture(models.Model):
     _inherit = 'hr.employee.departure'
 
-    do_unassign_company_car = fields.Boolean(
-        string="Release Company Car",
-        default=lambda self: self.env.user.has_group('fleet.fleet_group_user'),
-    )
-
     def action_register(self):
         super().action_register()
-        self.filtered('do_unassign_company_car')._unassign_company_car()
+        self._unassign_company_car()
 
     def _unassign_company_car(self):
         """Find all fleet.vehicle.assignation.log records that link to the employee, if there is no

--- a/addons/hr_fleet/models/hr_version.py
+++ b/None
@@ -1,9 +0,0 @@
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-from odoo import fields, models
-
-
-class HrVersion(models.Model):
-    _inherit = 'hr.version'
-
-    departure_do_unassign_company_car = fields.Boolean(related="departure_id.do_unassign_company_car")
